### DISCUSSION TITLE
Procedure to migrate (not upgrade) from LT 2.1.9 to newest version?

### DISCUSSION THREAD

**Discussion by fthommen** [Leantime/leantime#1393] [Q&A] [answered]
Hi,

before I start bumping into walls I wanted to ask, if there is an "official" way to migrate an LT 2.1.9 installation to the newest version. An in-place upgrade is not possible for technical reasons (PHP 8 installation on the system doesn't make sense). So we plan to set up a completely new system with the newest LT version. Will it work to export and re-import the LT database or are there additional steps to consider, because e.g. the database scheme has changed?

Thanks a lot

**Comment by TareqK** [comment, marked as answer]
I've done this before, basically just move the database and user files to an empty server, then install leantime and run the upgrades. It worked fine for me. 

If you want to be very conservative, you can move both your DB/User files and your currently installed version of leantime to a new server, and then run the updater.

**Reply by fthommen** [reply]
Thank you @TareqK . This would mean to install version 2.1.9 first, then upgrade? However I am sorely missing any information on upgrading, backing up and restoring on https://docs.leantime.io.

I will probably rather try your second suggestion, but would still need some information on how to upgrade.

**Reply by TareqK** [reply]
You can simply run the upgradeLeantime.sh script on your webhost if you are using the zip installation, or changing the version of the docker container you are using if using the docker version

**Reply by fthommen** [reply]
👍  @TareqK , I had not had a look at the files on the filesystem so far. I'm not using the docker version. However I have now copied the database files, recreated the Leantime database user with correct permissions on the new server, copied the current content of the leantime folder to the new server, but when I access the page, leantime seems to think this is a new site and displays the initial account creation page. When I try to create an initial account, I get the following error message:

<img width="496" alt="image" src="https://user-images.githubusercontent.com/279232/218280600-ff0f8909-186c-4531-bdcf-edc9d93637fd.png">

Something still seems to be missing, but I currently don't understand, what it is...

**Reply by fthommen** [reply]
Never mind. Instead of copying the database files I have now correctly dumped and reimported the database and I can now log in and see our current entries.  Then I did an update:

1. I downloaded Leantime release 2.3.6
2. Unzipped the content over the current files ("replace all")
3. Executed `bash ./updateLeantime.sh` only to see, that I already was at the newest version :-):
```
# bash ./updateLeantime.sh 

    _                 _   _                             
   | |   ___ __ _ _ _| |_(_)_ __  ___                   
   | |__/ -_) _` | ' \  _| | '  \/ -_)                  
   |____\___\__,_|_||_\__|_|_|_|_\___|      _           
                     | | | |_ __  __| |__ _| |_ ___ _ _ 
                     | |_| | '_ \/ _` / _` |  _/ -_) '_|
                      \___/| .__/\__,_\__,_|\__\___|_|  
                           |_|                          


For correct operation of this script, please ensure you have zip, unzip, wget and curl installed.

You are already up to date with v2.3.6!
# 
```

I assume I did manually what the script would have done :-). However there hadn't been an upgrade script at this before unzipping version 2.3.6.  Thanks for the help

**Reply by fthommen** [reply]
btw, 2.3.6 looks much better than 2.1.9 and finally there is a todo-overview which spans all projects. Thanks a lot
